src/main/java/org/folio/service/caches/JobProfileSnapshotCache.java

```diff
@@ -0,0 +1,72 @@
+package org.folio.service.caches;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.json.Json;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.rest.jaxrs.model.ProfileSnapshotWrapper;
+import org.folio.rest.util.OkapiConnectionParams;
+import org.folio.rest.util.RestUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.apache.http.HttpStatus;
+
+import com.github.benmanes.caffeine.cache.AsyncCache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+@Component
+public class JobProfileSnapshotCache {
+
+  private static final Logger LOGGER = LogManager.getLogger();
+
+  @Value("${orders.profile-snapshot-cache.expiration.time.seconds:3600}")
```

Comment:
Something rubs me the wrong way about this configuration name. I think it is the hypen and how long it is. Maybe `orders.cache.jobprofile.expiration.seconds`

Reply:
agree. completed.